server/src/main/java/org/opensearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -164,6 +175,10 @@ public String[] concreteIndexNames(ClusterState state, IndicesOptions options, I
         return concreteIndexNames(context, request.indices());
     }
 
+    public List<String> concreteSystemIndices(String... concreteIndices) {
```

Comment:
Actually, thinking about this API...there's nothing here specifically about "concrete" indices. The caller can pass in anything. I'm thinking we should either expose the system index patterns for the caller to do whatever it would like with them, or else change this to something like "matchesSystemIndexPattern()".

Reply:
I like the sound of `matchesSystemIndexPattern()`, making this change now.

Reply:
@andrross Pushed another commit to allow the caller to pass `indexExpressions`. This method now effectively acts as a filter to filter out the list of `indexExpressions` passed to the method and returns the expressions that match a system index (pattern).